scheduler/src/cook/mesos.clj

```diff
@@ -214,10 +217,12 @@
                                         (sched/create-datomic-scheduler
                                           mesos-datomic-conn
                                           (fn set-or-create-framework-id [framework-id]
-                                            (curator/set-or-create
-                                              curator-framework
-                                              zk-framework-id
-                                              (.getBytes (-> framework-id mesomatic.types/pb->data :value) "UTF-8")))
+                                            (let [value (-> framework-id mesomatic.types/pb->data :value)]
```

Comment:
Should this be guarded by a condition that checks that `framework-id-promise` has not been realized and also have an assertion/log if it has been delivered that the `framework-id` is the same value?